### PR TITLE
[1822PNW] City tiles for Portland and Seattle, bug fix for closing mi…

### PR DESCRIPTION
**Pull request by outsidePasser**
…nors

### CODE REVIEW

File: lib/engine/game/g_1822_pnw/game.rb [hunk 1]
```diff
@@ -696,31 +696,19 @@ def upgrades_to?(from, to, special = false, selected_company: nil)
...
-            company.owner.spend(payment, bank)
-          else
-            @log << "#{company.owner.name} cannot afford #{format_currency(payment)} for #{company.name}"
-            close_p16
+          if associated_minor?(corporation)
+            major = associated_major(corporation)
+            hex_by_id(corporation.coordinates).tile.cities[0].add_reservation!(major)
+            @log << "#{major.name} reservation takes the place of #{corporation.name}"
+          elsif regional_railway?(corporation)
+            company = company_by_id("M#{corporation.id}")
```
Comment (crericha) [inline lib/engine/game/g_1822_pnw/game.rb:706]: helper method for the magic

File: lib/engine/game/g_1822_pnw/game.rb [hunk 2]
```diff
@@ -768,6 +762,10 @@ def unassociated_minors
           @corporations.select { |c| c.floated? && c.type == :minor && !MINOR_ASSOCIATIONS.include?(c.id) }
         end
 
+        def regionals
+          @corporations.select { |c| regional_railway?(c) }
```
Comment (crericha) [inline lib/engine/game/g_1822_pnw/game.rb:766]: Cache this?

File: lib/engine/game/g_1822_pnw/game.rb (hunk 1 above)
Comment (outsidePasser) [reply lib/engine/game/g_1822_pnw/game.rb:706]: Made method "company_id_from_corp_id" for this

File: lib/engine/game/g_1822_pnw/game.rb (hunk 2 above)
Comment (outsidePasser) [reply lib/engine/game/g_1822_pnw/game.rb:766]: @corporations can change, so didn't cache it.  Added comment indicating that.

File: lib/engine/game/g_1822_pnw/game.rb [hunk 3]
```diff
@@ -662,20 +588,34 @@ def reorder_players(_order = nil)
...
+          # company.id == 'P9' || company.id == 'P17' || company.id == 'P18'
+        end
+
+        def choices_entities
+          []
         end
 
         def must_remove_town?(entity)
-          entity.id == 'P12' || entity.id == 'P13'
+          entity.id == 'P7' || entity.id == 'P8'
```
Comment (crericha) [inline lib/engine/game/g_1822_pnw/game.rb:618]: %w[P7 P8].include?(entity.id)

**Review by crericha — APPROVED** [review]

**Pull request merged**